scipy/stats/_binned_statistic.py

```diff
@@ -331,46 +447,60 @@ def binned_statistic_dd(sample, values, statistic='mean',
     if not callable(statistic) and statistic not in known_stats:
         raise ValueError('invalid statistic %r' % (statistic,))
 
+    # `Ndim` is the number of dimensions (e.g. `2` for `binned_statistic_2d`)
+    # `Dlen` is the length of elements along each dimension.
     # This code is based on np.histogramdd
     try:
-        # Sample is an ND-array.
-        N, D = sample.shape
+        # `sample` is an ND-array.
+        Dlen, Ndim = sample.shape
     except (AttributeError, ValueError):
-        # Sample is a sequence of 1D arrays.
+        # `sample` is a sequence of 1D arrays.
         sample = np.atleast_2d(sample).T
-        N, D = sample.shape
+        Dlen, Ndim = sample.shape
+
+    # Store initial shape of `values` to preserve it in the output
+    values = np.asarray(values)
+    inValuesShape = list(values.shape)
```

Comment:
minor stylistic point, but we tend to avoid CamelCase for variable names. I'd use ``input_shape`` or something like that.